sui_programmability/examples/nfts/sources/DiscountCoupon.move

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module NFTs::DiscountCoupon {
+    use Sui::Coin;
+    use Sui::SUI::SUI;
+    use Sui::NFT::{Self, NFT};
+    use Sui::TxContext::{Self, TxContext};
+
+    /// Discount coupon NFT.
+    struct DiscountCoupon has store {
+        // coupon issuer
+        issuer: address,
+        // percentage discount [1-100]
+        discount: u64,
```

Comment:
I believe `u8` would fit here better.

Reply:
It could be, that's a leftover since originally discount was price related (not %), but then we ended up on this.